iceoryx_hoofs/source/posix_wrapper/thread.cpp

```diff
@@ -49,5 +50,111 @@ ThreadName_t getThreadName(pthread_t thread) noexcept
     return ThreadName_t(cxx::TruncateToCapacity, tempName);
 }
 
+cxx::expected<ThreadError> ThreadBuilder::create(cxx::optional<Thread>& uninitializedThread,
+                                                 const Thread::callable_t& callable) noexcept
+{
+    if (!callable)
+    {
+        std::cerr << "The thread cannot be created with an empty callable." << std::endl;
```

Comment:
Could you please use `LogError` instead of `std::cerr` in the whole file. You gain access by including `iceoryx_hoofs/internal/log/hoofs_logging.hpp`

Reply:
@mossmaurice thanks for the little research. 

I think we gain a lot of flexibility when we using our logger. `std::cout` prints only to console (of course you can redirect it but then you maybe also redirect the `cout` messages of top level applications which they do not want).